adapters/rubicon/rubicon_test.go

```diff
@@ -411,6 +414,132 @@ func TestParseSizes(t *testing.T) {
 	}
 }
 
+func TestMASAlgorithm(t *testing.T) {
+	SIZE_ID := getTestSizes()
+	type output struct {
+		primary int
+		alt     []int
+		ok      bool
+	}
+	type testStub struct {
+		input  []openrtb.Format
+		output output
+	}
+
+	testStubs := []testStub{
+		{
+			[]openrtb.Format{
+				SIZE_ID[2],
+				SIZE_ID[9],
+			},
+			output{2, []int{9}, false},
+		},
+		{
+			[]openrtb.Format{
+
+				SIZE_ID[9],
+				SIZE_ID[15],
+			},
+			output{15, []int{9}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[2],
+				SIZE_ID[15],
+			},
+			output{15, []int{2}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[15],
+				SIZE_ID[9],
+				SIZE_ID[2],
+			},
+			output{15, []int{2, 9}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[10],
+				SIZE_ID[9],
+				SIZE_ID[2],
+			},
+			output{2, []int{10, 9}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[33],
+				SIZE_ID[8],
+				SIZE_ID[15],
+			},
+			output{15, []int{33, 8}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[33],
+				SIZE_ID[8],
+				SIZE_ID[9],
+				SIZE_ID[2],
+			},
+			output{2, []int{33, 8, 9}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[33],
+				SIZE_ID[8],
+				SIZE_ID[9],
+			},
+			output{9, []int{33, 8}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[33],
+				SIZE_ID[8],
+				SIZE_ID[2],
+			},
+			output{2, []int{33, 8}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[33],
+				SIZE_ID[2],
+			},
+			output{2, []int{33}, false},
+		},
+		{
+			[]openrtb.Format{
+				SIZE_ID[8],
+			},
+			output{8, []int{}, false},
+		},
```

Comment:
This is a good set of tests -- suggest adding one more which is a list of sizes that's not 2,9,15.

Reply:
That one's already covered in https://github.com/prebid/prebid-server/blob/36b03e07d07876d78532fb33fc19e0123269a7fe/adapters/rubicon/rubicon_test.go#L815